File: app/src/main/java/com/leaguechampions/dagger/component/AppComponent.java
package com.leaguechampions.dagger.component;

import com.leaguechampions.activity.ChampionDetailsActivity;
import com.leaguechampions.activity.ChampionsActivity;
import com.leaguechampions.activity.SettingsActivity;
import com.leaguechampions.dagger.module.AppModule;
import com.leaguechampions.dagger.module.NetworkModule;
import com.leaguechampions.module.DataModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, NetworkModule.class, DataModule.class})
public interface AppComponent {
    void inject(ChampionsActivity activity);
    void inject(ChampionDetailsActivity activity);
    void inject(SettingsActivity activity);
}

File: app/src/main/java/com/leaguechampions/injection/component/AppComponent.kt
package com.leaguechampions.injection.component

import android.content.Context
import com.leaguechampions.features.champions.injection.ChampionComponent
import com.leaguechampions.features.champions.injection.ChampionModule
import com.leaguechampions.injection.module.NetworkModule
import com.leaguechampions.injection.module.StorageModule
import com.leaguechampions.injection.viewmodel.ViewModelModule
import com.leaguechampions.libraries.core.injection.CoreDataModule
import com.leaguechampions.libraries.core.injection.DataModule
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    StorageModule::class,
    NetworkModule::class,
    DataModule::class,
    ViewModelModule::class,

    ChampionModule::class,

    CoreDataModule::class
])
interface AppComponent {
    @Component.Factory
    interface Factory {
        fun create(@BindsInstance context: Context): AppComponent
    }

    fun championComponent(): ChampionComponent.Factory
}
